tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3762 @@
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 2</strong> to denote a <i>Metadata Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data Length</td>
+                                <td>uint32</td>
+                                <td>
+                                    Length of the following data field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data</td>
+                                <td>ByteArray</td>
+                                <td>Binary data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                    <details>
+                        <summary>Tag Block</summary>
+                        <blockquote>
+                            Defines an indexation tag to which the output can be indexed by additional node plugins.
+                        </blockquote>
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 3</strong> to denote a <i>Tag Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag Length</td>
+                                <td>uint8</td>
+                                <td>
+                                    Length of the following tag field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag</td>
+                                <td>ByteArray</td>
+                                <td>Binary indexation data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                </td>
+            </tr>
+        </table>
+    </details>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+- `Amount` field must fulfill the dust protection requirements.
```

Comment:
For all outputs, maybe this needs a bit of clarification. I think The actual value of the amount is checked against zero in the Transaction TIP but not in this one, which would suggest that an Output with a 0 amount is valid in itself, until it is put in a transaction. I would say it's invalid in the first place.

Reply:
Indeed, it should be invalid.